synapse/push/baserules.py

```diff
@@ -238,6 +238,25 @@ def make_base_prepend_rules(
         ],
         "actions": ["dont_notify"],
     },
+    # Enable notifications for replies without fallback
+    {
+        "rule_id": "global/override/.im.nheko.msc3664.reply",
+        "conditions": [
+            # Only send notification if the reply is to your message
+            {
+                "kind": "im.nheko.msc3664.related_event_match",
+                "key": "sender",
+                "rel_type": "m.in_reply_to",
+                "pattern_type": "user_id",
+                "_id": "_reply",
```

Comment:
I think I just got caught out by this - the `_id` field is [used to cache conditon results in the bulk evaluator](https://github.com/matrix-org/synapse/blob/e24ff8ebe3d4119d377355402245947f7de61c00/synapse/push/bulk_push_rule_evaluator.py#L277-L283) (seems to be the only place it's used) and so is unsafe to use when matching against the current user ID ([the cache is utilised on the room/event, rather than room/event/user, level](https://github.com/matrix-org/synapse/blob/e24ff8ebe3d4119d377355402245947f7de61c00/synapse/push/bulk_push_rule_evaluator.py#L210)).